components/ws-manager/pkg/manager/create.go

```diff
@@ -266,6 +266,28 @@ func (m *Manager) createDefiniteWorkspacePod(startContext *startWorkspaceContext
 		prefix = "ghost"
 	case api.WorkspaceType_IMAGEBUILD:
 		prefix = "imagebuild"
+		// mount self-signed gitpod CA certificate to ensure
+		// buildkit can push images to the in-cluster registry
+		workspaceContainer.VolumeMounts = append(workspaceContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      "gitpod-ca-certificate",
+				MountPath: "/usr/local/share/ca-certificates/gitpod-ca.crt",
+				SubPath:   "ca.crt",
+				ReadOnly:  true,
+			},
+			corev1.VolumeMount{
+				Name:      "image-builder-mk3-config",
+				MountPath: "/etc/buildkit/buildkitd.toml",
+				SubPath:   "buildkitd.toml",
+				ReadOnly:  true,
+			},
+		)
+
+		// signal workspacekit to mount required files to push in-cluster registry
+		workspaceContainer.Env = append(workspaceContainer.Env, corev1.EnvVar{
+			Name:  "GITPOD_WORKSPACEKIT_BIND_MOUNTS",
+			Value: `["/etc/buildkit/buildkitd.toml","/usr/local/share/ca-certificates/gitpod-ca.crt"]`,
```

Comment:
If we wanted this behaviour to be permanent, I'd rather we these paths to the list of "[well known locations](https://github.com/gitpod-io/gitpod/blob/main/components/workspacekit/cmd/rings.go#L470)" in workspacekit.